### PR TITLE
Switch from webpki-roots to rustls-native-certs

### LINKED ISSUE #184: cmsis-pack crate does not respect local configuration of root certificates
`Cache` is unable to download packs in enterprise environments with "transparent" TLS-terminating middleboxes:

```
$ pack-manager cache packs
Caching all packs; this will take some time. Press Control-C to quit.
20:57:14 [WARN] Sending fatal alert BadCertificate
20:57:14 [ERROR] https://sadevicepacksprodus.blob.core.windows.net/idxfile/index.pidx: error trying to connect: invalid certificate: UnknownIssuer
```

`reqwest` [supports](https://docs.rs/reqwest/0.10.10/reqwest/#optional-features) bundling Mozilla's root certificates (`rusttls-tls`, i.e., `rust-tls-webpki-roots`) or using the platform's native certificate store (`rustls-tls-native-roots`). See [`rustls-native-certs`](https://github.com/rustls/rustls-native-certs#should-i-use-this-or-webpki-roots) for a comparison of the two approaches.

Consider switching from `rust-tls-webpki-roots` to `rustls-tls-native-roots` to support enterprise environments with "transparent" TLS-terminating middleboxes.

### PR DESCRIPTION
**Pull request by claymation**
Fixes #184.

### CODE REVIEW

File: rust/cmsis-pack/Cargo.toml [hunk 1]
```diff
@@ -22,7 +22,7 @@ rustc-demangle = "0.1.16"
 serde = { version = "1.0.118", features = ["derive"] }
 serde_json = "1.0"
 tokio = { version = "1.0", features = ["macros", "rt"] }
-reqwest = { version = "0.11.0", default_features = false, features = ["rustls-tls", "trust-dns", "stream"] }
+reqwest = { version = "0.11.0", default_features = false, features = ["rustls-tls-native-certs", "trust-dns", "stream"] }
```
Comment (mathias-arm) [inline rust/cmsis-pack/Cargo.toml:25]: I think the feature in `reqwest` is called `rustls-tls-native-roots` and uses the `rustls-tls-native-certs` crate.

**Review by mathias-arm — CHANGES_REQUESTED** [review]

**Review by mathias-arm — APPROVED** [review]
Not a fan of "transparent" TLS proxies, but honoring the system trust preferences seems like a good thing. This change will probably need additional cross-platform testing.

**Comment by kand198** [comment]
I'd like to lend a hand in getting this merged, if possible.

I have tested these changes on `Ubuntu 22.04.2 LTS on Windows 10 x86_64` (WSL2).

What further testing would you like to see?

**Comment by mathias-arm** [comment]
@claymation, @kand198: I rebased the PR. If you could test that it still fixes the issue and that there are no regressions (in particular on Windows).

**Comment by kand198** [comment]
I have re-tested under Windows 10 natively (non-WSL), as well as Ubuntu 22.04 under WSL2. Everything builds and runs fine. I can update packs from behind my work proxy. Tests pass.

**Pull request merged**